### ISSUE TITLE
Doesn't work on Ubuntu 17.10 (artful)

### ISSUE THREAD

**Issue by sapphirecat** [JRaspass/Cache-Memcached-Fast#12]
The Ubuntu repo's version 0.23, and the latest 0.25 tarball, both exhibit the same behavior: they don't think they successfully connected, and they don't perform any operations.

This happens against real memcached, and 0.23 was also observed to fail when sending to a Memcached::Server.

* server_versions returns `{}`
* set and get return `undef`
* tcpdump sees 6 (3-way setup, 3-way teardown) packets of 0 bytes each, no other traffic
* `make test` for 0.25 skips everything because it thinks there's no memcached server
* `Cache::Memcached` works fine (tested on real memcached only)
* Everything is fine on Ubuntu 17.04 (zesty, shipped with version 0.23).

The memcached server can be located with `sudo netstat -lntp` and `ps`; its cmdline is `/usr/bin/memcached -m 64 -p 11211 -u memcache -l 127.0.0.1`

AFAICT, the only changes in memcached or C::M::F between the Ubuntu releases is "No-change rebuild against libevent-2.1-6" for memcached, and the updated Perl version (5.26).

**Comment by sapphirecat** [comment]
I've been trying to characterize this better.  I installed Perlbrew, and found the package fails under Artful with brewed 5.24.3 and 5.26.1.

I built with `make OPTIMIZE='-O2 -Wall -Wextra -g'` to possibly get some more help; the command output with 5.24.3 is attached in [cmf-524.log](https://github.com/JRaspass/Cache-Memcached-Fast/files/1427600/cmf-524.log).

I'm attaching my testing script as well, 
[mcd-test.zip](https://github.com/JRaspass/Cache-Memcached-Fast/files/1427611/mcd-test.zip)

**Comment by sapphirecat** [comment]
Full test results of version 0.25:

* ubuntu zesty, system perl 5.24.1: OK
* ubuntu artful, system perl 5.26.0: fail
* ubuntu artful, brewed perl 5.24.3: fail
* ubuntu artful, brewed perl 5.26.1: fail
* debian stretch, brewed perl 5.26.1: OK
* debian buster, brewed perl 5.26.1: OK
* debian buster, brewed perl 5.26.0: OK
* debian buster, system perl 5.26.0: OK

"OK" represents `make test` fully passing; "fail" represents all the tests being skipped due to no connection.

Debian stretch is stable, and buster is testing.

Unless further info is requested, I think I'm going to quit worrying about it at this point.  It's pretty clear Ubuntu _specifically_ has broken something.  I can switch our code to vanilla `Cache::Memcached` which works on artful.

**Comment by sapphirecat** [comment]
For the record, **it appears fixed for Ubuntu 18.04 LTS (Bionic Beaver).**  I was able to confirm that both their `libcache-memcached-fast-perl` and version 0.25 installed with `cpanm` run the previously attached test script properly.  I double-checked the cpanm build log, and it connected successfully and passed tests properly there, too.

I used the beta2 server CD, with updates through today installed.

I also discovered [an Ubuntu bug](https://bugs.launchpad.net/ubuntu/+source/libcache-memcached-fast-perl/+bug/1733655) with no more information than I was able to find last October.
